ngraph/python/src/pyngraph/dict_attribute_visitor.hpp

```diff
@@ -32,114 +33,96 @@ namespace util
+        void on_adapter(const std::string& name,
+                        ngraph::ValueAccessor<std::string>& adapter) override;
+        void on_adapter(const std::string& name, ngraph::ValueAccessor<int8_t>& adapter) override;
+        void on_adapter(const std::string& name, ngraph::ValueAccessor<int16_t>& adapter) override;
+        void on_adapter(const std::string& name, ngraph::ValueAccessor<int32_t>& adapter) override;
+        void on_adapter(const std::string& name, ngraph::ValueAccessor<int64_t>& adapter) override;
+        void on_adapter(const std::string& name, ngraph::ValueAccessor<uint8_t>& adapter) override;
+        void on_adapter(const std::string& name, ngraph::ValueAccessor<uint16_t>& adapter) override;
+        void on_adapter(const std::string& name, ngraph::ValueAccessor<uint32_t>& adapter) override;
+        void on_adapter(const std::string& name, ngraph::ValueAccessor<uint64_t>& adapter) override;
+        void on_adapter(const std::string& name, ngraph::ValueAccessor<float>& adapter) override;
+        void on_adapter(const std::string& name, ngraph::ValueAccessor<double>& adapter) override;
+        void on_adapter(const std::string& name,
+                        ngraph::ValueAccessor<std::vector<std::string>>& adapter) override;
+        void on_adapter(const std::string& name,
+                        ngraph::ValueAccessor<std::vector<int8_t>>& adapter) override;
+        void on_adapter(const std::string& name,
+                        ngraph::ValueAccessor<std::vector<int16_t>>& adapter) override;
+        void on_adapter(const std::string& name,
+                        ngraph::ValueAccessor<std::vector<int32_t>>& adapter) override;
+        void on_adapter(const std::string& name,
+                        ngraph::ValueAccessor<std::vector<int64_t>>& adapter) override;
+        void on_adapter(const std::string& name,
+                        ngraph::ValueAccessor<std::vector<uint8_t>>& adapter) override;
+        void on_adapter(const std::string& name,
+                        ngraph::ValueAccessor<std::vector<uint16_t>>& adapter) override;
+        void on_adapter(const std::string& name,
+                        ngraph::ValueAccessor<std::vector<uint32_t>>& adapter) override;
+        void on_adapter(const std::string& name,
+                        ngraph::ValueAccessor<std::vector<uint64_t>>& adapter) override;
+        void on_adapter(const std::string& name,
+                        ngraph::ValueAccessor<std::vector<float>>& adapter) override;
+        void on_adapter(const std::string& name,
+                        ngraph::ValueAccessor<std::vector<double>>& adapter) override;
+
+        void on_adapter(const std::string& name,
+                        ngraph::ValueAccessor<std::shared_ptr<ngraph::Function>>& adapter) override;
 
     protected:
         const py::dict& m_attributes;
+        std::unordered_map<std::string, std::shared_ptr<ngraph::Variable>>& m_variables;
```

Comment:
The `m_variables` is populated by the deserializer class but the deserialized values are not used anywhere. Can you explain why we're populating this map at all?

Reply:
They are used. We need to set exactly the same version of Variable to Assign/ReadValue ops. 
ngraph/python/src/pyngraph/dict_attribute_visitor.cpp file, line 140:
```
if (!m_variables.count(variable_id))
{
    m_variables[variable_id] = std::make_shared<ngraph::Variable>(ngraph::VariableInfo{
        ngraph::PartialShape::dynamic(), ngraph::element::dynamic, variable_id});
}
a->set(m_variables[variable_id]);
```
m_variables map is used to store created Variables by variable_id (std::string). 
If Variable with the variable_id exists in the map, then we use it, otherwise we create a new one and add it to the map.